backend/rust-bindings/src/lib.rs

```diff
@@ -28,6 +28,21 @@ impl FFIOption {
     }
 }
 
+fn assign_result<F: FnOnce() -> Result<T>, T: ToString>(target: *mut *mut c_char, f: F) -> bool {
```

Comment:
Maybe add docstring like "Compute result using the provided callback, convert it into a C string and assign it to the provided target. The returned bool indicates whether the computation was successful or not." Or whatever may be more accurate.

I don't think the function name exactly conveys the behavior, but to not having this turn into excessive nitting, I will leave it with that.